tests/test_ml.py

```diff
@@ -547,13 +712,30 @@ def test_model_format_validation_errors(self, model_format):
         "abc",
         {},
         [],
-        True
+        True,
+        ml.CoreMlFormat.from_dict(COREML_FORMAT_JSON),
```

Comment:
I'd just say that this also could be problematic. If `from_dict()` is buggy and throws, the whole test suite will fail. A better approach would be to extract these into a separate test case:

```
@parametrize('model_source_json', json_args_list)
def test_tflite_format_coreml_source(model_source_json):
  model_source = CoreMlFormat.from_dict(model_source_json)
  # Rest of the test case
```

You don't necessarily have to fix it here and now. Just something to keep in mind.